oe_theme.theme

```diff
@@ -840,6 +840,45 @@ function oe_theme_preprocess_paragraph__oe_content_row__variant_inpage_navigatio
   $variables['links'] = $links;
 }
 
+/**
+ * Implements hook_preprocess_paragraph() for paragraph--oe-text-feature-media.html.twig.
+ */
+function oe_theme_preprocess_paragraph__oe_text_feature_media(array &$variables): void {
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['paragraph'];
+
+  $cacheability = CacheableMetadata::createFromRenderArray($variables);
+
+  // Extract the media if present.
+  if ($paragraph->get('field_oe_media')->isEmpty()) {
+    return;
+  }
+  /** @var \Drupal\file\Plugin\Field\FieldType\FileItem $media_item */
+  $media_item = $paragraph->get('field_oe_media')->first();
+  /** @var \Drupal\media\Entity\Media $media */
+  $media = $media_item->get('entity')->getValue();
+  // Caches are handled by the formatter usually. Since we are not rendering
+  // the original render arrays, we need to propagate our caches to the
+  // paragraph template.
+  $cacheability->addCacheableDependency($media);
+  // Extract the image if present.
+  if ($media->get('oe_media_image')->isEmpty()) {
+    return;
```

Comment:
Here you return but you did not apply the `cacheability` to the variables so if the media object behind later *gets* an image, it will stay cached without.